### ISSUE TITLE
default creation of identity prevents proper working of plugins new_user_identity and new_user_dialog

### ISSUE THREAD

**Issue by f1-outsourcing** [roundcube/roundcubemail#7085]
the new_user_dialog is selecting the wrong identity, not the identity from new_user_identity which was created from ldap data.
Thus if you enable new_user_identity, this default created should be disabled. 

Where/how can I disable this?

**Comment by alecpl** [comment]
Both plugins should use the same (default) identity, so I'm not sure I understand.

**Comment by f1-outsourcing** [comment]
the idea of the 'new_user_identity' is that you get a correct identity from in my case an ldap server. 
the idea of the 'new_user_dialog' is that you get a popup if this identity being added is not complete.

What happens when you create a new account
new account logs in
rc is making new user A
rc is making new identity A (on estimated values)

Now if you have the plugins new_user_identity and new_user_dialog enabled.
rc gives popup with data if identity A
rc adds from ldap identity B

So the user edits a shitty parameters in a popup of identity A while the correct ones are in identity B. This does not make sense.  That is opposite of the idea of getting good values from ldap. 

I think you should add some conditions to this line 694, maybe a config entry where you can choose to have this default identity be created. Because you do not want it, when you are getting it from somewhere else. And plugins get confused.


```
    684                     $record['name'] = $user_name != $record['email'] ? $user_name : '';
    685                 }
    686
    687                 $record['user_id']  = $user_id;
    688                 $record['standard'] = $standard;
    689
    690                 $plugin = $rcube->plugins->exec_hook('identity_create',
    691                     array('login' => true, 'record' => $record));
    692
    693                 if (!$plugin['abort'] && $plugin['record']['email']) {
    694                     //$rcube->user->insert_identity($plugin['record']);
    695                 }
    696                 $standard = 0;
    697             }
    698         }
    699         else {
    700             rcube::raise_error(array(
    701                 'code' => 500,
    702                 'type' => 'php',
    703      
```

**Comment by alecpl** [comment]
I have a feeling that all your issues are related to the ldap setup and content. So, could you show your config and the ldap record content?

**Comment by alecpl** [comment]
No feedback. I'm not able to reproduce.

**Comment by alecpl** [comment]
Re-open, as I was finally able to reproduce some issue.

If lookup_user_name() returns:
```
array(
    'user_name' => "User Name",
    'user_email' => "test1@alec.pl",
    'email_list' => array("test2@alec.pl"),
)
```
There will be only one identity created, the one with test2@alec.pl address. And because it's the only one identity (so it's the default identity) it will be used in new_user_dialog dialog.

The proper outcome should be two identities (for both addresses), and test1@alec.pl being the default, and this one should be displayed in new_user_dialog dialog.

**Comment by f1-outsourcing** [comment]
I was about to update rc and noticed some stuff I edited, I am not sure if or how this could be of assistance.

```
diff -r '--exclude=temp' '--exclude=logs' webmail.test.com/plugins/new_user_identity/new_user_identity.php webmail2.test.com/plugins/new_user_identity/new_user_identity.php
23,24c23
<         //$this->add_hook('user_create', array($this, 'lookup_user_name'));
<         //$this->add_hook('user_create', array($this, 'login_after'));
---
>         $this->add_hook('user_create', array($this, 
> 'lookup_user_name'));
36d34
<                 $user_organization = is_array($results->records[0]['organization']) ? $results->records[0]['organization'][0] : $results->records[0]['organization'];
39d36
<                 $args['user_organization']  = $user_organization;
68,71c65
< 	$defid = $this->rc->user->get_identity(null);
< 
< 	
<         if ($this->ldap || !$this->rc->config->get('new_user_identity_onlogin') || !empty($defid)) {
---
>         if ($this->ldap || 
> !$this->rc->config->get('new_user_identity_onlogin')) {
74,83d67
< 	//error_log(">>> executing login_after");
< 	//error_log(">>> defid: ".implode($defid));
< 	//error_log(">>> empty: ".empty($defid));
< 
< 	//delete the default created identity
<         // this does not work, because the popup is hooking already on the first user creation
< 	//$defidt = $this->rc->user->get_identity(null);
< 	//error_log(">>> identity id to delete ".$defidt['identity_id']);
<         //error_log(">>>  ".implode($defidt));
< 	//$this->rc->user->delete_identity($defidt['identity_id']);
104,106c88
<                     'organization' => $ldap_entry['user_organization'],
<                     //'name'     => $ldap_entry['user_name']
<                     'name'     => '' 
---
>                     'name'     => $ldap_entry['user_name']
diff -r '--exclude=temp' '--exclude=logs' webmail.test.com/program/lib/Roundcube/rcube_user.php webmail2.test.com/program/lib/Roundcube/rcube_user.php
694c694
<                     //$rcube->user->insert_identity($plugin['record']);
---
>                     $rcube->user->insert_identity($plugin['record']);

diff -r '--exclude=temp' '--exclude=logs' webmail.test.com/public_html/plugins/new_user_identity/new_user_identity.php webmail2.test.com/public_html/plugins/new_user_identity/new_user_identity.php
23,24c23
<         //$this->add_hook('user_create', array($this, 'lookup_user_name'));
<         //$this->add_hook('user_create', array($this, 'login_after'));
---
>         $this->add_hook('user_create', array($this, 
> 'lookup_user_name'));
36d34
<                 $user_organization = is_array($results->records[0]['organization']) ? $results->records[0]['organization'][0] : $results->records[0]['organization'];
39d36
<                 $args['user_organization']  = $user_organization;
68,71c65
< 	$defid = $this->rc->user->get_identity(null);
< 
< 	
<         if ($this->ldap || !$this->rc->config->get('new_user_identity_onlogin') || !empty($defid)) {
---
>         if ($this->ldap || 
> !$this->rc->config->get('new_user_identity_onlogin')) {
74,83d67
< 	//error_log(">>> executing login_after");
< 	//error_log(">>> defid: ".implode($defid));
< 	//error_log(">>> empty: ".empty($defid));
< 
< 	//delete the default created identity
<         // this does not work, because the popup is hooking already on the first user creation
< 	//$defidt = $this->rc->user->get_identity(null);
< 	//error_log(">>> identity id to delete ".$defidt['identity_id']);
<         //error_log(">>>  ".implode($defidt));
< 	//$this->rc->user->delete_identity($defidt['identity_id']);
104,106c88
<                     'organization' => $ldap_entry['user_organization'],
<                     //'name'     => $ldap_entry['user_name']
<                     'name'     => '' 
---
>                     'name'     => $ldap_entry['user_name']

```

**Comment by alecpl** [comment]
Hopefully fixed with 8cf8a4738176. Although looking at it again I'm not 100% sure. We should dig into what really happens in `lookup_user_name()` if this is still an issue.

**Comment by f1-outsourcing** [comment]
@alecpl  Just installed from scratch 1.6.2 (with new_user_dialog and new_user_identity)

looks like the ldap is getting some records. 

```
[26-Aug-2023 20:36:01 +0000]: <6sggpq2u> Executing search with return attributes: array (
  0 => 'sendmailmtakey',
  1 => 'sendmailmtakey',
  2 => 'sendmailmtakey',
  3 => 'o',
  4 => 'sendmailmtamapvalue',
  5 => 'objectClass',
)
[26-Aug-2023 20:36:01 +0000]: <6sggpq2u> S: 2 record(s) found
```

But the dialog popup has nothing that comes from the ldap when a new user logs in.
